src/lab4.py

```diff
@@ -30,19 +27,18 @@ def __str__(self):
     
     def __repr__(self):
         return f"PreciousStone({self.__name}, {self.__carats}, {self.__price})"
-
-    def __del__(self):
-        print(f"PreciousStone {self.__name} is being deleted")
+    
+    def details(self):
+        return f"{self}"
 
 def main():
-    stone1 = PreciousStone(1.5, 5000, "Diamond")
-    stone2 = PreciousStone(2.0, 10000, "Ruby")
-    stone3 = PreciousStone(0.8, 2000, "Sapphire")
+    diamond = PreciousStone(1.5, 5000, "Diamond")
+    ruby = PreciousStone(2.0, 10000, "Ruby")
+    sapphire = PreciousStone(0.8, 2000, "Sapphire")
 
-    for stone in [stone1, stone2, stone3]:
-        print(stone)
-        print("Public Number Field:", stone.public_number_field)
-        print("Public String Field:", stone.public_string_field)
+    stones = [Diamond, Ruby, Sapphire]
```

Comment:
you changed the names of objects, so you should also change it when you make a list